### DISCUSSION TITLE
Errors in  SMTP logs

### DISCUSSION THREAD

**Discussion by jaydeepnp** [postalserver/postal#2530] [Help with using Postal]
Hi,

Can anyone  please help with below ERRORS in smtp log.

smtp_1      | [9] [2023-07-06T13:55:18.218] ERROR -- : An error occurred while accepting a new client.
smtp_1      | [9] [2023-07-06T13:55:18.218] ERROR -- : Errno::EPIPE: Broken pipe
smtp_1      | [9] [2023-07-06T13:55:18.218] ERROR -- : /opt/postal/app/lib/postal/smtp_server/server.rb:109:in `write'
smtp_1      | [9] [2023-07-06T13:55:18.218] ERROR -- : /opt/postal/app/lib/postal/smtp_server/server.rb:109:in `print'
smtp_1      | [9] [2023-07-06T13:55:18.218] ERROR -- : /opt/postal/app/lib/postal/smtp_server/server.rb:109:in `block (2 levels) in run_event_loop'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /opt/postal/app/lib/postal/smtp_server/server.rb:86:in `select'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /opt/postal/app/lib/postal/smtp_server/server.rb:86:in `block in run_event_loop'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /opt/postal/app/lib/postal/smtp_server/server.rb:84:in `loop'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /opt/postal/app/lib/postal/smtp_server/server.rb:84:in `run_event_loop'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /opt/postal/app/lib/postal/smtp_server/server.rb:287:in `run'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /opt/postal/app/lib/tasks/postal.rake:15:in `block (2 levels) in <top (required)>'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `block in execute'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `each'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:281:in `execute'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/sentry-ruby-5.8.0/lib/sentry/rake.rb:24:in `execute'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:219:in `block in invoke_with_call_chain'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `synchronize'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:199:in `invoke_with_call_chain'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/task.rb:188:in `invoke'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:160:in `invoke_task'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block (2 levels) in top_level'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `each'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:116:in `block in top_level'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:125:in `run_with_threads'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:110:in `top_level'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:83:in `block in run'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:186:in `standard_exception_handling'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/lib/ruby/gems/3.2.0/gems/rake-13.0.6/lib/rake/application.rb:80:in `run'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /opt/postal/app/bin/rake:4:in `<top (required)>'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/lib/bundler/cli/exec.rb:58:in `load'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/lib/bundler/cli/exec.rb:58:in `kernel_load'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/lib/bundler/cli/exec.rb:23:in `run'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/lib/bundler/cli.rb:492:in `exec'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/lib/bundler/vendor/thor/lib/thor.rb:392:in `dispatch'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/lib/bundler/cli.rb:34:in `dispatch'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/lib/bundler/vendor/thor/lib/thor/base.rb:485:in `start'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/lib/bundler/cli.rb:28:in `start'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/exe/bundle:45:in `block in <top (required)>'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/lib/bundler/friendly_errors.rb:117:in `with_friendly_errors'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /usr/local/bundle/gems/bundler-2.4.9/exe/bundle:33:in `<top (required)>'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /opt/postal/app/bin/bundle:3:in `load'
smtp_1      | [9] [2023-07-06T13:55:18.219] ERROR -- : /opt/postal/app/bin/bundle:3:in `<main>'

How we can  remove this errors.

Regards,
Jaydeep

**Comment by willpower232** [comment]
> Errno::EPIPE: Broken pipe

This usually means there was a problem with the network connection, are you actually having problems sending messages or is this just a message you're seeing a lot of?

Unfortunately having an email server on the internet means you get a lot of interested strangers poking around!

**Reply by jaydeepnp** [reply]
Will,

Thanks a lot for your response.

All working fine  as far as sending / receiving  emails.

I am seeing this errors on regular interval on SMTP logs so wanted to check if by any way we can  remove / resolve this errors.

Yes Agree email server on internet meaning  many unwanted connections to server.  From the logs it seems all related to client (ruby/gems etc) connection I think.
 
Regards,
Jaydeep
